File: UdemyCoursePy/src/mx/irving/stacksandqueuesandmore/Queue.py
class Queue(object):

	def __init__(self):
		self.items = []
		
	def enqueue(self, item):
		self.items.append(item)
		
	def dequeue(self):
		return self.items.pop(0)
	
	def isEmpty(self):
		return self.items == []
	
	def size(self):
		return len(self.items)

		
queue = Queue()
print(queue.size())
queue.enqueue("ONE")
queue.enqueue(2)
queue.enqueue(True)
print(queue.size())
print(queue.dequeue())
print(queue.size())
print(queue.dequeue())
print(queue.size())
print(queue.isEmpty())
print(queue.dequeue())
print(queue.isEmpty())

File: UdemyCourseKt/src/mx/irving/queuesandstacks/Queue.kt
class Queue {

	val data = mutableListOf<Any>()
	
	fun enqueue(item: Any) {
		data.add(item)
	}
	
	fun dequeue(): Any {
		return data.removeAt(0)
	}
	
	fun isEmpty(): Boolean {
		return data.isEmpty()
	}
	
	fun size(): Int {
		return data.size
	}
}


fun main() {
	val queue = Queue()
	println(queue.isEmpty())
	queue.enqueue("ONE")
	queue.enqueue(2)
	queue.enqueue(true)
	println(queue.isEmpty())
	println(queue.size())
	println(queue.dequeue())
	println(queue.dequeue())
	println(queue.size())
	println(queue.isEmpty())
	println(queue.dequeue())
	println(queue.isEmpty())
}
